torchbenchmark/models/sam/__init__.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# This software may be used and distributed according to the terms of the GNU General Public License version 3.
+
+from ...util.model import BenchmarkModel
+from .build_sam import sam_model_registry
+from .predictor import SamPredictor
+from PIL import Image
+import numpy as np
+import cv2
+from torchbenchmark.tasks import COMPUTER_VISION
+import torch
+import os
+
+    
+class Model(BenchmarkModel):
+    task = COMPUTER_VISION.SEGMENTATION
+    DEFAULT_EVAL_BSIZE = 32
+    
+    def __init__(self, test, device, jit=False, batch_size=1, extra_args=[]):
+        super().__init__(test=test, device=device, jit=jit, batch_size=batch_size, extra_args=extra_args)
+        
+        # Checkpoint options are here https://github.com/facebookresearch/segment-anything#model-checkpoints
+        data_folder = os.path.join(os.path.dirname(os.path.abspath(__file__)), '.data')
+        sam_checkpoint = os.path.join(data_folder, 'sam_vit_h_4b8939.pth')
+        model_type = "vit_h"
+
+        self.model = sam_model_registry[model_type](checkpoint=sam_checkpoint)
+        self.model.to(device=device)   
+        data_folder = os.path.join(os.path.dirname(os.path.abspath(__file__)), '.data')
+
+        image_path = os.path.join(data_folder, 'truck.jpg')
+        self.image = cv2.imread(image_path)
+        self.image = cv2.cvtColor(self.image, cv2.COLOR_BGR2RGB)     
+        self.sample_image  = torch.randn((3, 256, 256)).to(device)
+
+   
+    def get_module(self):
+        example_input = [
```

Comment:
Looks like the code for `get_module()` is different from `eval()`, I am wondering why is that?

This will actually cause different performance numbers reported by `run.py` and `torchbench.py` (https://github.com/pytorch/pytorch/blob/main/benchmarks/dynamo/torchbench.py).

Reply:
Yeah so basically one is just benchmarking the nn module and the other is an end to end model with a predictor

IMO both are interesting but I'm more interested in the full end to end setup but I couldn't get get_module to work with something that wasn't an nn module